### DISCUSSION TITLE
Failed to get installed location of package 'Microsoft.VCLibs.140.00.UWPDesktop_14.0.30035.0_x64__8wekyb3d8bbwe' because it's not staged

### DISCUSSION THREAD

**Discussion by ilovepumpkin** [microsoft/WindowsAppSDK#3533] [Q&A]
Hello,

We have a Desktop application. In its installer, there is one .misxbundle file and some dependent .appx packages. After running the following command in Power Shell console, we got the following error.

```
Add-AppProvisionedPackage -Online -PackagePath .\MyApplication.msixbundle -DependencyPackagePath .\Dependencies\x64\Microsoft.VCLibs.x64.14.00.appx,.\Dependencies\x64\Microsoft.VCLibs.x64.14.00.Desktop.appx,.\Dependencies\x64\Microsoft.WindowsAppRuntime.1.0.msix -SkipLicense
Add-AppProvisionedPackage : The parameter is incorrect.
At line:1 char:1
+ Add-AppProvisionedPackage -Online -PackagePath .\MyApplication. ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  + CategoryInfo     : NotSpecified: (:) [Add-AppxProvisionedPackage], PSArgumentException
  + FullyQualifiedErrorId : Microsoft.Dism.Commands.AddAppxProvisionedPackageCommand
```

In  C:\Windows\Logs\DISM\dism.log, we found the following error:
```
2023-03-24 12:19:11, Error                 DISM   DISM Appx Provider: PID=11400 TID=17712 onecore\admin\appmodel\utilities\provisionhelper\msixpackageadapter.cpp(647)\AppxProvider.dll!00007FFB68EB5100: (caller: 00007FFB68EB522E) ReturnHr(1) tid(4530) 80070057 The parameter is incorrect.
    Msg:[Failed to get installed location of package 'Microsoft.VCLibs.140.00.UWPDesktop_14.0.30035.0_x64__8wekyb3d8bbwe' because it's not staged] 
 - CAppxManager::ResultLoggingCallback
```

I tried to google the error message "Failed to get installed location of package 'Microsoft.VCLibs.140.00.UWPDesktop_14.0.30035.0_x64__8wekyb3d8bbwe' because it's not staged" but found nothing. The interesting thing is I could install Microsoft.VCLibs.140.00.UWPDesktop successfully using the command "Add-AppxPackage .\Dependencies\x64\Microsoft.VCLibs.x64.14.00.Desktop.appx".

What does the error "Failed to get installed location of package 'Microsoft.VCLibs.140.00.UWPDesktop_14.0.30035.0_x64__8wekyb3d8bbwe' because it's not staged" mean? How did it happen? What should I do to make it work in the first "Add-AppProvisionedPackage" command?

**Comment by DrusTheAxe** [comment]
Run at an elevated prompt

`powershell -c get-appxprovisionedpackage -online`

and see if any Microsoft.VCLibs.140.00.UWPDesktop package is provisioned. If not, or there is but not x64 and an equal or higher version, then you're trying to provision a package without its dependency being available. To succeed you need to provision a compatible Microsoft.VCLibs.140.00.UWPDesktop package explicitly, or provide it with your `Add-AppxProvisionedPackage` cmmand via the `-DependencyPackagePath` option.

https://learn.microsoft.com/en-us/powershell/module/dism/add-appxprovisionedpackage?view=windowsserver2022-ps
